Yoast/javascript
# Improve Arabic text analysis by stemming prefix combinations

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Adds prefixes وبال, مست, تنن, است, نست, ولل, كال, فال, بال, وال, لن, سن, ست, ات, وس, فل, فب, فس, لل, ول, وب, ال to the list of ignored prefixes for Arabic. When the Arabic keyphrase is preceded by one of these prefixes, the basic keyphrase forms will also be created.
* [Yoast SEO Free] Makes it possible to recognize forms of keyphrases in Arabic when they are preceded by certain multi-letter prefixes, e.g. والقبعة, فسحرية, or وبالمفتاح.

## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

* Test the following functionality in Free

* Set your site to Arabic

* As your keyphrase, set the word طائر.

* Add a Arabic text with at least 150 words, e.g. using https://generator.lorem-ipsum.info/_arabic or Arabic Wikipedia.

* In the text, add the form وبالطائر. The keyphrase density assessment should show 1 keyphrase occurrence. When clicking on the marker for the assessment, وبالطائر should be marked.

* Switch around the keyphrase and the word form.

* The keyphrase density assessment should still show 1 keyphrase occurrence.

* Switch the form in the text to a form with another prefix, e.g. كالطائر or وللطائر.

* The keyphrase density assessment should still show 1 keyphrase occurrence.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-530
